src/python/pants/engine/streaming_workunit_handler.py

```diff
@@ -138,7 +138,7 @@ class WorkunitsCallbackFactory:
     rule. See #11354 for discussion of that limitation.
     """
 
-    callback_factory: Callable[[], WorkunitsCallback]
+    callback_factory: Callable[[], WorkunitsCallback] | None
```

Comment:
Maybe:
```suggestion
    callback_factory: Callable[[], WorkunitsCallback | None]
```
...instead? The idea here was that the callback needs to be created per run. Usually the entire factory will be re-created (if the inputs to the `@rule` that creates it have changed), but it might be useful for the factory to be able to decide per-run based on ... some other property?

Reply:
Yea, I like that. 👍🏽